### PR TITLE
Expand and update docs on monorepo package publishing

### PR DESCRIPTION
**Pull request by jsnajdr**
Today with @tyxla we did a Zoom call with a walkthrough of the NPM publishing process. This documentation update in one of the outcomes. It describes the process and all the gotchas and options in much more detail and hopefully it will be useful.

### CODE REVIEW

File: docs/monorepo.md [hunk 1]
```diff
@@ -133,22 +133,70 @@ Note that if you're building with Webpack, you may need to turn off [`resolve.sy
...
+or
+```
+git tag "@automattic/components@1.0.0"
+```
+
+Now `lerna publish from-git` will offer to publish only the packages that have a matching Git tag on the current `HEAD` revision.
+
+The rest of the workflow is exactly the same as in the `from-package` case.
+
+### Getting NPM permissions to publish in the `@automattic` scope
```
Comment (tyxla) [inline docs/monorepo.md]: Should this be moved above as it's a prerequisite for being able to publish?

File: docs/monorepo.md [hunk 2]
```diff
@@ -133,22 +133,70 @@ Note that if you're building with Webpack, you may need to turn off [`resolve.sy
...
-1. Pat yourself on the back, you published!
+We use [Lerna](https://lernajs.io/) and its `publish` command to publish the monorepo packages to NPM registry. Please do not use regular [`npm publish`](https://docs.npmjs.com/cli/publish) within a package to publish an individual package; `npx` has issues using this flow.
+
+### Make sure changelogs and `package.json` versions are up to date
+
+For all packages that you want to publish, make sure that their `package.json` versions are bumped. Decide carefully whether you want to publish a patch, a minor or a major update of the package. Be mindful about [semantic versioning](https://semver.org/).
+
+Make sure that the `CHANGELOG.md` document contains up-to-date information, with the `next` heading replaced with the version number that you are about to publish.
+
+Create PRs with the necessary changes and merge them to `master` before publishing. Lerna will add a `gitHead` field to each published package's `package.json`. That field contains the hash of the Git commit that the package was published from. It's better if this commit hash is a permanent one from the `master` branch, rather than an ephemeral commit from a local branch.
```
Comment (tyxla) [inline docs/monorepo.md]: 👍

**Review by tyxla — APPROVED** [review]
This looks awesome, thank you @jsnajdr! Just left a minor optional suggestion, but feel free to 🚢 either way!

File: docs/monorepo.md [hunk 3]
```diff
@@ -133,22 +133,70 @@ Note that if you're building with Webpack, you may need to turn off [`resolve.sy
...
+
+Now make sure you're logged in at this point, we're going to publish 🚀: `npm whoami`, `npm login`. Enter your username, password, and the OTP code.
+
+Craft the following command, we'll add `--yes` to skip prompts and save OTP cycle time. `--dist-tag next` is optional, use it when publishing unstable versions.
+
+Wait for your NPM OTP (one time password) cycle to start -- if your NPM account is correctly set up for maximum security, you'll need to enter the OTP code when publishing, even though you've already logged in with another OTP. Write the code into the command and publish:
+
+```
+NPM_CONFIG_OTP=[YOUR_OTP_CODE] npx lerna publish --dist-tag next from-package --yes
+```
```
Comment (sirreal) [inline docs/monorepo.md]: I think we can remove the OTP cycle time and `NPM_CONFIG_OTP=…` bits here. `lerna` now asks for OTP at the right time.

File: docs/monorepo.md [hunk 4]
```diff
@@ -133,22 +133,70 @@ Note that if you're building with Webpack, you may need to turn off [`resolve.sy
...
+git tag "@automattic/calypso-build@6.1.0"
+```
+or
+```
+git tag "@automattic/components@1.0.0"
+```
+
+Now `lerna publish from-git` will offer to publish only the packages that have a matching Git tag on the current `HEAD` revision.
+
+The rest of the workflow is exactly the same as in the `from-package` case.
```
Comment (sirreal) [inline docs/monorepo.md]: Do we want to suggest `from-git`? It seemed very problematic when we tried to use it early on. Maybe we should suggest adding `private:true` to `package.json` temporarily.

Comment (tyxla) [reply docs/monorepo.md]: Seems like it's the most straightforward way to publish a single package. The `private:true` one seems even more like a hack 😉 What was problematic about it before?

File: docs/monorepo.md (hunk 3 above)
Comment (jsnajdr) [reply docs/monorepo.md]: Alright. I can confirm that Lerna/NPM asks interactively if you don't set the environment variable.

File: docs/monorepo.md (hunk 4 above)
Comment (jsnajdr) [reply docs/monorepo.md]: @sirreal I use `from-git` every time I'm publishing, haven't run into any issues so far, and I can recommend this workflow 🙂 If there are any risks, let's describe them.

**Comment by jsnajdr** [comment]
Pushed an update that addresses review feedback:
- put the section about NPM accounts and the `automattic` org before we start describing `lerna publish`
- describe the interactive flow (with fewer CLI) options by default. Describe the non-interactive options in a premium section
- describe the `--dist-tag next` option in a dedicated section. Keep the main line simple.

**Review by sirreal — APPROVED** [review]
Thanks for updating these!

**Pull request merged**